lib/logqueue-fifo.c

```diff
@@ -267,6 +268,7 @@ log_queue_fifo_push_tail(LogQueue *s, LogMessage *msg, const LogPathOptions *pat
            * input thread finishes */
 
           main_loop_worker_register_batch_callback(&self->qoverflow_input[thread_id].cb);
+          self->qoverflow_input[thread_id].cb.user_data = log_queue_ref(s);
           self->qoverflow_input[thread_id].finish_cb_registered = TRUE;
```

Comment:
self vs s:
- If I use self, then I have to use &self->super or casting it: (LogQueue *)self 
- But the input parameter is s and s is a LogQueue.
So I thought, that using is the more simple way (My opionon is that as clean as the using self).
But I you want I will use &self->super